src/Symfony/Component/Security/Http/Authenticator/Passport/PassportInterface.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\Passport;
+
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\BadgeInterface;
+
+/**
+ * A Passport contains all security-related information that needs to be
+ * validated during authentication.
+ *
+ * A passport badge can be used to add any additional information to the
+ * passport.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @experimental in 5.1
+ */
+interface PassportInterface
+{
+    /**
+     * Adds a new security badge.
+     *
+     * A passport can hold only one instance of the same security badge.
+     * This method replaces the current badge if it is already set on this
+     * passport.
+     *
+     * @return $this
+     */
+    public function addBadge(BadgeInterface $badge): self;
```

Comment:
I know it looks minor ... but if we are applying the passport metaphor ... we should rename "badge" as "stamp". The reason is that passports don't have "badges" but "stamps". See https://en.wikipedia.org/wiki/Passport_stamp

Reply:
I agree. Only thing is that this might be confused with the stamps of the messenger component.

Reply:
I disagree, because a "passport stamp" means something different. Don't you get a stamp to show that HAVE entered a country and when? So it's something you get *after* they decide to allow you in? Badges are different: they often define other things that need to be checked. Immigration would never "check for a stamp to see if they also need to perform further authentication checks". That's why I think badge is better - it has no meaning in a passport system... but "stamp" has the *wrong* meaning.

Reply:
Ryan is right! Let's keep the badges then. Thanks!